src/coreclr/nativeaot/BuildIntegration/NativeAOT.py

```diff
@@ -0,0 +1,82 @@
+"""
+LLDB pretty printer for NativeAOT
+
+Add to ~/.lldbinit:
+
+command script import {path-to-ilcompiler-package}/build/NativeAOT.lldb.py
```

Comment:
`path-to-ilcompiler-package` is not constant. It changes every month.

Would it be better to include this in our LLDB debugger extension?

Reply:
It has same release cadence as SDK. File is next to NativeAOT.natvis.

`dotnet-debugger-extensions` and `dotnet-sos` have different release cadence. Also, `debugger extension` is not open source and unavailable in VMR distro builds; the only option there is dotnet-sos, so we will need to add it to two packages that way.

Reply:
NativeAOT.natvis is bundled into the .pdb file. People do not need to worry about explicitly setting a path to it. This python script is not like .natvis.

Yes, .NET debuggers are on a different shipping train and licensing plan than .NET itself.

Reply:
Looks like the Python script could be embedded into symbols and LLDB would pick it up but I only see it implemented for mach-O: https://github.com/llvm/llvm-project/blob/705e27c23474f3177670a791b5b54eefedee0cd8/lldb/source/Plugins/Platform/MacOSX/PlatformDarwin.cpp#L202-L318

From what I see in the source code, the experience would be that LLDB will notify there's a debugging script and one then needs to make a gesture to allow it.

I like the approach from https://github.com/dotnet/runtime/issues/115406#issuecomment-2864162604 - the only unfortunate thing about that one is that the LLDB change to read the embedded formatters was only made 5 months ago so it'll take years until Linux distros pick that up.

Reply:
It can be implemented as SOS sub-command in C++ which gives us full control.